### ISSUE TITLE
systemd-nspawn -D <container> prints `fails to create directory` warning without SELinux

### ISSUE THREAD

**Issue by wsw70** [systemd/systemd#3711]
### Submission type
- [x] Bug report
- [ ] Request for enhancement (RFE)
### systemd version the issue has been seen with

> 229
### Used distribution

> Ubuntu 16.04 (xenial)
### In case of bug report: Expected behaviour you didn't see

> (no error messages)
### In case of bug report: Unexpected behaviour you saw

> error message `Failed to create directory /var/lib/machines/testcont/sys/fs/selinux: Read-only file system`
### In case of bug report: Steps to reproduce the problem

```
# mkdir /var/lib/machines
# cd /var/lib/machines
# debootstrap xenial testcont http://fr.archive.ubuntu.com/ubuntu/
(output from the creation of the base system, no errors)
I: Retrieving InRelease
I: Checking Release signature
I: Valid Release signature (key id 790BC7277767219C42C86F933B4FE6ACC0B21F32)
(...)
I: Configuring ureadahead...
I: Configuring resolvconf...
I: Base system installed successfully.
# systemd-nspawn -D testcont
Spawning container testcont on /var/lib/machines/testcont.
Press ^] three times within 1s to kill container.
Failed to create directory /var/lib/machines/testcont/sys/fs/selinux: Read-only file system
Failed to create directory /var/lib/machines/testcont/sys/fs/selinux: Read-only file system
root@testcon #
```

**Comment by poettering** [comment]
hmm, i figure your kernel doesn't have selinux enabled, does it? I figure we need to be more permissive with expecting that /sys/fs/selinux exists in sysfs as empty directory that selinuxfs can be mounted on

**Comment by wsw70** [comment]
This is a default installation of Ubuntu 16.04 so you are correct, `selinux` is not enabled.

**Comment by martinpitt** [comment]
Right, ubuntu uses apparmor by default. You can boot with `selinux=1 security=selinux` to enable and switch to selinux.

**Comment by martinpitt** [comment]
What I don't understand is why this works everywhere else. I run nspawn all the time on Ubuntu 16.04 (with the default kernel, with AppArmor), and that Just Works. You also seem to run ubuntu in the guest, so there's nothing magic there.

**Comment by martinpitt** [comment]
.. oh, and we also use nspawn in the systemd CI, without any magical options. It works fine there as well.

**Comment by wsw70** [comment]
@martinpitt : you do not have any warning? In my case, beside the warning, everything works fine.

**Comment by mbiebl** [comment]
It's a warning for me, not a fatal error (on Debian unstable)

**Comment by martinpitt** [comment]
@wsw70 : Oh, I indeed have that warning, I just never paid attention. It sounded like this was a fatal error, sorry.

**Comment by poettering** [comment]
Fixed by #3748.
